sos/plugins/openstack_instack.py

```diff
@@ -28,6 +30,41 @@ def setup(self):
         self.add_copy_spec("/home/stack/instackenv.json")
         self.add_copy_spec("/home/stack/undercloud.conf")
 
+        run_cmds = False
+        for os_var in ['OS_USERNAME', 'OS_PASSWORD', 'OS_TENANT_NAME']:
+            if os_var not in os.environ:
+                self.soslog.warning("%s not found in environment variables"
+                                    " which is required to run commands"
+                                    % (os_var))
+            else:
+                run_cmds = True
+
+        if run_cmds:
+            # get status of overcloud stack and resources
+            self.add_cmd_output(
+                "openstack stack show overcloud",
+                suggest_filename="openstack_stack_show_overcloud")
+            self.add_cmd_output(
+                "openstack stack resource list -n 10 overcloud",
+                suggest_filename="openstack_stack_resource_list-n10_overcloud")
+
+            # get details on failed deployments
+            cmd = "openstack stack resource list -f value -n 5 overcloud"
+            deployments = self.call_ext_prog(cmd)['output']
+            for deployment in deployments.splitlines():
+                if 'FAILED' in deployment:
+                    check = [
+                        "OS::Heat::StructuredDeployment",
+                        "OS::Heat::SoftwareDeployment"]
+                    if any(x in deployment for x in check):
+                        deployment = deployment.split()[1]
+                        cmd = "openstack software deployment show " \
+                                "--long %s" % (deployment)
+                        self.add_cmd_output(
+                            cmd,
```

Comment:
Why does this need to be on a newline? Only lists inside `[..]` use the bracket-then-newline style in sos unless there's no space on the line.

If that makes the line with the `kwarg` harder to format I'd suggest using a variable for the file name - building them in-line is always a bit tricky to format readably since the argument name is already so long.